pkg/daemon/server/service/rater/timestamped_counts.go

```diff
@@ -85,57 +82,19 @@ func (tc *TimestampedCounts) PodReadCountSnapshot() map[string]map[string]float6
 	return counts
 }
 
-// PodDeltaCountSnapshot returns a copy of the podName to partition delta mapping
-// it's used to ensure the returned map is not modified by other goroutines
-func (tc *TimestampedCounts) PodDeltaCountSnapshot() map[string]map[string]float64 {
-	tc.lock.RLock()
-	defer tc.lock.RUnlock()
-	counts := make(map[string]map[string]float64)
-	for k, v := range tc.podPartitionDelta {
-		counts[k] = v
-	}
-	return counts
-}
-
 // IsWindowClosed returns whether the window is closed
 func (tc *TimestampedCounts) IsWindowClosed() bool {
 	tc.lock.RLock()
 	defer tc.lock.RUnlock()
 	return tc.isWindowClosed
 }
 
-// CloseWindow closes the window and calculates the delta by comparing the current pod counts with the previous window
-func (tc *TimestampedCounts) CloseWindow(prev *TimestampedCounts) {
-	podReadCount := tc.PodReadCountSnapshot()
-	var prevPodReadCount map[string]map[string]float64
-	if prev == nil {
-		prevPodReadCount = make(map[string]map[string]float64)
-	} else {
-		prevPodReadCount = prev.PodReadCountSnapshot()
-	}
-	podPartitionDelta := make(map[string]map[string]float64)
-
-	for podName, partitionReadCounts := range podReadCount {
-		prevPartitionReadCounts := prevPodReadCount[podName]
-		for partitionName, count := range partitionReadCounts {
-			prevCount := prevPartitionReadCounts[partitionName]
-			// delta will be equal to count in case of restart
-			delta := count
-			if count >= prevCount {
-				delta = count - prevCount
-			}
-			if _, ok := podPartitionDelta[podName]; !ok {
-				podPartitionDelta[podName] = make(map[string]float64)
-			}
-			podPartitionDelta[podName][partitionName] = delta
-		}
-	}
-
-	// finalize the window by setting isWindowClosed to true and delta to the calculated value
+// CloseWindow closes the window
+func (tc *TimestampedCounts) CloseWindow() {
```

Comment:
we can remove this since the window is not closed only for the latest timestamp (last element of the queue) which we skip for rate calculation for the same reason.